content/docs/2.9/scalers/pulsar.md

```diff
@@ -28,22 +29,42 @@ triggers:
 - `subscription` - Name of the topic subscription
 - `msgBacklogThreshold` - Average target value to trigger scaling actions. (default: 10)
 - `activationMsgBacklogThreshold` - Target value for activating the scaler. Learn more about activation [here](./../concepts/scaling-deployments.md#activating-and-scaling-thresholds).(Default: `0`, Optional)
+- `authModes` - a comma separated list of authentication modes to use. Default `""`. Valid options are `bearer`, `tls`, and `basic`. Note that `tls,bearer` or `tls,basic` are valid combinations and would indicate mutual TLS to secure the connection and then `bearer` or `basic` headers should be added to the HTTP request.
 
 ### Authentication Parameters
 
-  If TLS is required you should set `tls` to `enable`. If required for your Pulsar configuration, you may also provide a `ca`, `cert` and `key`. `ca`, `cert` and `key` must be specified together.
+  The `authModes` are specified on the `metadata` field of the `ScaledObject`, as shown above. The associated configuration parameters are specified in the `TriggerAuthentication` spec.  
 
+**Bearer Auth**
+
+When configuring Bearer Authentication (Token Auth), configure the following:
+
+- `bearerToken`: This token will be sent as a header in the form `Authorization: Bearer <token>`. 
+
+**Basic Auth**
+When configuring Basic Authentication, configure the following:
+
+- `username`: the username
+- `password`: the password (optional)
 
 **TLS:**
 
-- `tls`: Optional. To enable SSL auth for Pulsar, set this to `enable`. If not set, TLS for Pulsar is not used, Your shoule set this key to trigger metadata.
-- `ca`: Certificate authority file for TLS client authentication. 
+When configuring mutual TLS authentication, configure the following:
+
+- `ca`: The trusted root Certificate authority used to validate the server's certificate.
 - `cert`: Certificate for client authentication.
 - `key`: Key for client authentication. 
 
-### Example
 
-Your pulsar cluster no TLS auth:
+### TLS with custom CA Certificates
+
+When configuring a trusted root CA that is not well known, it is sufficient to specify the `ca` field on the `TriggerAuthentication` resource. See [Bearer Token with TLS via custom trusted CA Certificate](#bearer-token-with-tls-via-custom-trusted-ca-certificate) for an example.
+
+Before 2.9.0, it was necessary to configure `tls: enable` on the `ScaledObject`. That was removed in a backwards compatible way, so you can remove that field now.
```

Comment:
I'm not sure if we should specify this here because this is 2.9 indeed, so changes are clear. We can/should add a deprecation message in the log, but I'm not sure if here is the best place.
@kedacore/keda-docs ?

Reply:
I think that it is okay to keep it here, probably marked as 
```
> **Note:**
>
```

I am not sure if we want to have a deprecation message in the log? as the config will still work, it won't be changed in the future

Reply:
do we want to support the old `tls` approach having the new one?

Reply:
@JorTurFer I am not sure I follow. @michaeljmarshall mentioned that the change is done in a backwards compatible way. Ie -> before it was needed to specify `tls: enable`, now you don't have to do so, but you can keep the setting there. No change in the functionality in the end.

Reply:
Maybe I misunderstood the change, but I supposed this will be the new "desired" way for the scaler. Am I right? @michaeljmarshall ?

Reply:
@kedacore/keda-docs , we should merge this soon because the feature was merged by error before having the docs ready.
Do you have any thoughts you'd like to share?

Reply:
I don't see what the problem is with this PR to be honest.

The security fix by @michaeljmarshall seems unrelated to this and does not provide any configuration options anyway; but I might be missing something.

Reply:
@JorTurFer @tomkerkhove so what about this?

Reply:
Resolving